packages/adp-tooling/src/preview/change-handler.ts

```diff
@@ -58,10 +85,19 @@ export function isAddXMLChange(change: CommonChangeProperties): change is AddXML
 export function addXmlFragment(basePath: string, change: AddXMLChange, fs: Editor, logger: Logger): void {
     const { fragmentPath } = change.content;
     const fullPath = join(basePath, DirName.Changes, fragmentPath);
-
+    const templateConfig = fragmentTemplateDefinitions[change.selector?.templateName ?? ''];
```

Comment:
I don't know how this will affect the change if there is another property in the selector object called `templateName `. This needs to be consulted with RTA colleagues. Did you test if this works?

Reply:
Yes, it worked, but now this code refactored according to Kevin recommendations. content.templateName is used instead of selector.